packages/pino-log-enricher/lib/createFormatter.js

```diff
@@ -39,6 +39,18 @@ module.exports = function createFormatter(newrelic) {
         }
         return obj
       }
+    },
+    hooks: {
+      logMethod(inputArgs, method, level) {
+        // We'll try to use level labels for the metric name, but if
+        // they don't exist, we'll default back to the level number.
+        const levelLabel = this.levels.labels[level] || level
```

Comment:
I did have this problem while testing but that's because I called this function incorrectly. But it only has a single calling site and it does properly bind `this` there to the pino object.

So I think as part of the pino API, I can always assume `this`, `this.levels` and `this.levels.labels` are all defined.

Reply:
it may seem like overkill but as an agent we don't like to crash a customer app. I'm not sure of a scenario where it could happen but it's just worth calling out

Reply:
I think I'll let this go for now. As we've discussed, the whole approach is kind of brittle to begin with because we're providing a whole config that could conflict with a customer-supplied config. My understanding is that we have plans to do the whole thing elsehow, so let's keep this for now until we have a better way to instrument Pino.